tests/drivers/flash/common/src/main.c

```diff
@@ -42,6 +42,20 @@
 #error "Unsupported configuraiton"
 #endif
 
+#define TEST_FLASH_START (DT_REG_ADDR(DT_CHOSEN(zephyr_flash)))
```

Comment:
hi @nordic-segl, I'm seeing failures on this test after these changes for some boards where the storage partition is in an external flash different from the internal flash where zephyr is executing (`zephyr,flash`). It seems that is the test now is wrongly using both the storage partition and the `zephyr,flash` area, or I'm misunderstanding the purpose of this. Could you please help me clarify what was the intention here so that I can get it fixed for our boards? thanks

Reply:
Hi,
It's not trivial for 'generic' driver to detect what are the correct address values for FLASH.

There is already
https://github.com/zephyrproject-rtos/zephyr/issues/77331

I'm afraid that my PR introduced regression on many targets.
Do You get failures in test_negative_flash_* or in other tests?

Reply:
I wanted to add negative tests for flash_erase(), flash_fill(), flash_flatten(), flash_read() and flash_write().
Tests expect negative error code returned by these functions when:
- memory address is "before" flash start (TEST_FLASH_START - 1);
- memory address is to high (TEST_FLASH_START + TEST_FLASH_SIZE);
- length of data to be read is too big (TEST_FLASH_SIZE - TEST_AREA_OFFSET + 1).
I assumed that flash start address and it's size can be read from zephyr_flash node in the DTS.

Reply:
> Do You get failures in test_negative_flash_* or in other tests?

Only in the test_negative_flash_*, the rest work fine on my target. The problem in my case is that TEST_FLASH_START/SIZE are obtained from the internal flash, but the flash under test is the external flash which address starts at 0 for the storage partition. I’m thinking if it’s not better for now to skip the negative tests until a more generic approach is found.

Reply:
I confirm that regressions are introduced on STM32 as well